packages/main/cypress/specs/DateTimePicker.cy.tsx

```diff
@@ -0,0 +1,303 @@
+					const timestamp = $el.attr("data-sap-timestamp") || "";
+					const date = new Date(parseInt(timestamp) * 1000);
+					selectedDate = `${String(date.getDate()).padStart(2, "0")}/${String(date.getMonth() + 1).padStart(2, "0")}/${date.getFullYear()}`;
+				})
+				.realClick();
+
+			cy.get("#ok").realClick();
+		});
+
+		cy.get("#dtSeconds")
+			.shadow()
+			.find("ui5-input")
+			.invoke("prop", "value")
+			.should(val => {
+				expect(val).to.include(selectedDate);
+			});
+	});
+
+	it("tests time controls for dtSeconds picker", () => {
+		const expectedClocksCount = 3;
+		const expectedPeriodCount = 1;
+
+		cy.mount(<DateTimePickerWithSeconds initialValue="13/04/2020, 03:16:16 AM" />);
+
+		cy.ui5DateTimePickerOpen("#dtSeconds");
+
+		cy.ui5DateTimePickerTimeSelectionClocksCount("#dtSeconds").then(clocksCount => {
+			expect(clocksCount).to.equal(expectedClocksCount,
+				"The picker should display 3 clocks for hours, minutes and seconds.");
+		});
+
+		cy.ui5DateTimePickerPeriodSegmentedButtonCount("#dtSeconds").then(periodCount => {
+			expect(periodCount).to.equal(expectedPeriodCount,
+				"The picker should display 1 period selector.");
+		});
+
+		cy.ui5DateTimePickerClose("#dtSeconds");
+	});
+
+	it("tests time controls for dtMinutes picker", () => {
+		const expectedClocksCount = 2;
+		const expectedPeriodCount = 1;
+
+		cy.mount(<DateTimePickerWithMinutes />);
+
+		cy.ui5DateTimePickerOpen("#dtMinutes");
+
+		cy.ui5DateTimePickerTimeSelectionClocksCount("#dtMinutes").then(clocksCount => {
+			expect(clocksCount).to.equal(expectedClocksCount,
+				"The picker should display 2 clocks for hours and minutes.");
+		});
+
+		cy.ui5DateTimePickerPeriodSegmentedButtonCount("#dtMinutes").then(periodCount => {
+			expect(periodCount).to.equal(expectedPeriodCount,
+				"The picker should display 1 period selector.");
+		});
+
+		cy.ui5DateTimePickerClose("#dtMinutes");
+	});
+
+	it("tests hours clock is active on picker open", () => {
+		cy.mount(<DefaultDateTimePicker />);
+		cy.ui5DateTimePickerOpen("#dt");
+
+		cy.ui5DateTimePickerGetPicker("#dt").within(() => {
+			cy.get("ui5-time-selection-clocks")
+				.shadow()
+				.find(`ui5-time-picker-clock[data-ui5-clock="hours"]`)
+				.invoke("prop", "active")
+				.should("equal", true);
+		});
+
+		cy.ui5DateTimePickerClose("#dt");
+	});
+
+	it("tests selection of 12:34:56 AM", () => {
+		cy.mount(<DateTimePickerWithSeconds initialValue="13/04/2020, 03:16:16 AM" />);
+
+		cy.ui5DateTimePickerOpen("#dtSeconds");
+
+		cy.ui5DateTimePickerGetPicker("#dtSeconds").within(() => {
+			cy.get("ui5-time-selection-clocks")
+				.shadow()
+				.find(`ui5-toggle-spin-button[data-ui5-clock="hours"]`)
+				.click();
+			cy.realType("123456a"); // select 12:34:56 AM
+			cy.get("#ok").click();
+		});
+
+		cy.get("#dtSeconds")
+			.shadow()
+			.find("ui5-input")
+			.invoke("prop", "value")
+			.should("equal", "13/04/2020, 12:34:56 AM");
+	});
+
+	it("tests selection of 12:34:56 PM", () => {
```

Comment:
This tests runs successfully standalone, but it is unstable when executed with the other tests.